sound/soc/sof/sof-audio.c

```diff
@@ -43,12 +43,36 @@ static int sof_kcontrol_setup(struct snd_sof_dev *sdev, struct snd_sof_control *
 	return ret;
 }
 
-static int sof_dai_config_setup(struct snd_sof_dev *sdev, struct snd_sof_dai *dai)
+static struct sof_ipc_dai_config* sof_dai_config_match(struct snd_sof_dev *sdev,
+						       struct snd_sof_dai *dai,
+						       struct sof_ipc_pcm_params *ipc_pcm)
+{
+	int i;
+	int num_conf = dai->num_conf;
+	int sample_rate = ipc_pcm->params.rate;
+
+	/* match for ssp sample rate and fsync for now, otherwise use default config */
+	for (i = 0; i < num_conf; i++) {
+		if (dai->dai_config[i].type == SOF_DAI_INTEL_SSP &&
+		    sample_rate == dai->dai_config[i].ssp.fsync_rate) {
+			dev_dbg(sdev->dev, "found dai config match at index %d\n", i);
+			return &dai->dai_config[i];
+		}
+	}
+
+	dev_dbg(sdev->dev, "using default dai config %d\n", dai->def_conf);
+	return &dai->dai_config[dai->def_conf];
+}
+
+static int sof_dai_config_setup(struct snd_sof_dev *sdev, struct snd_sof_dai *dai,
+				struct sof_ipc_pcm_params *ipc_pcm)
 {
 	struct sof_ipc_reply reply;
-	struct sof_ipc_dai_config *config = dai->dai_config;
+	struct sof_ipc_dai_config *config = NULL;
```

Comment:
is the NULL init required?

Reply:
no, one could just (optionally) move the assignment below here
```
struct sof_ipc_dai_config *config = sof_dai_config_match(sdev, dai, ipc_pcm);
```